### PR TITLE
Dialog Titles

### PR DESCRIPTION
**Pull request by cwmanning**
Optionally specify Dialog titles via `title` prop.

### CODE REVIEW

File: addons/dialog/index.jsx [hunk 1]
```diff
@@ -10,11 +10,23 @@ module.exports = React.createClass({
     }
   },
 
+  getTitle() {
+    let { title } = this.props
+    if (this.props.title) {
+      return (
+        <h3 className="col-dialog-title">
+          { this.props.title }
+        </h3>
```
Comment (nhunzaker) [inline addons/dialog/index.jsx]: What if the `h3` were customizable here with like a `headingLevel` or something?

Comment (cwmanning) [inline addons/dialog/index.jsx]: Good call, I was going to ask you if `h3` was purposeful.

File: addons/dialog/index.jsx [hunk 2]
```diff
@@ -6,7 +6,17 @@ module.exports = React.createClass({
...
-      className: 'col-dialog'
+      className: 'col-dialog',
+      headingLevel: 'h3'
+    }
+  },
+
+  getTitle() {
+    let { headingLevel, title } = this.props
+    if (title) {
+      return React.DOM[headingLevel]({
```
Comment (cwmanning) [inline addons/dialog/index.jsx]: this could error out if mis-used I guess. thoughts @nhunzaker?

Comment (nhunzaker) [inline addons/dialog/index.jsx]: Hmm. What about changing it to `titleComponent` and then doing:

```javascript
return React.createElement(titleComponent, { className: 'col-dialog-title' }, title)

Comment (nhunzaker) [inline addons/dialog/index.jsx]: (or whatever prop name you prefer)

Comment (nhunzaker) [inline addons/dialog/index.jsx]: `createElement` can accept a string value for a known entity (like `h3`) or a component definition

**Comment by nhunzaker** [comment]
:+1:

**Pull request merged**